### PR TITLE
Added error and notification handling

### PR DESCRIPTION
**Pull request by shanejonas**
image

this relies on https://github.com/MetaMask/iframe-execution-environment/pull/24

### CODE REVIEW

File: src/transports/IframeExecutionEnvironmentTransport.ts [hunk 1]
```diff
@@ -63,12 +63,35 @@ class IframeExecutionEnvironmentTransport extends Transport {
 
     this.commandStream = mux.createStream('command') as unknown as Duplex;
 
-    this.commandStream.on('data', (data: unknown) => {
+    this.commandStream.on('data', (data: any) => {
```
Comment (shanejonas) [inline src/transports/IframeExecutionEnvironmentTransport.ts:66]:
```suggestion
    this.commandStream.on('data', (data: unknown) => {
```

File: src/transports/IframeExecutionEnvironmentTransport.ts [hunk 2]
```diff
@@ -66,9 +66,32 @@ class IframeExecutionEnvironmentTransport extends Transport {
...
+      error: {
+        code: error.code,
+        message: error.message,
+        data: error.data,
+      },
+    });
+  }
+
+  private notificationHandler = (event: any) => {
+    window.parent.postMessage(event, '*');
```
Comment (Gudahtt) [inline src/transports/IframeExecutionEnvironmentTransport.ts:92]: What does `window.parent` refer to here? I thought this was running _outside_ of the iframe execution environment :thinking:

Comment (shanejonas) [reply src/transports/IframeExecutionEnvironmentTransport.ts:92]: `window.parent` here is inspector.open-rpc.org 

context:

inspector.open-rpc.org => iframe postmessage => this code (which itself makes an iframe and talks over  postmessage/object multiplex).

**Review by Gudahtt — DISMISSED** [review]
LGTM! I don't fully understand the `this.transportRequestManager.transportEventChannel` API but since that isn't public API, understanding that well would require reading through the entire rpc client codebase. I'm going to trust that you understand what's going on with that part.

File: src/transports/IframeExecutionEnvironmentTransport.ts [hunk 3]
```diff
@@ -66,9 +66,32 @@ class IframeExecutionEnvironmentTransport extends Transport {
...
+      this.notificationHandler,
+    );
+    this.transportRequestManager.transportEventChannel.addListener(
+      'error',
+      this.errorNotificationHandler.bind(this),
+    );
     return true;
   }
 
+  private errorNotificationHandler(error: any) {
```
Comment (shanejonas) [inline src/transports/IframeExecutionEnvironmentTransport.ts:80]:
```suggestion
  private errorNotificationHandler(error: Error) {
```

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**